From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6nke=20Holz?= <soenke.holz@serenityos.org>
Date: Wed, 8 Jan 2025 21:10:05 +0100
Subject: [PATCH] liblzma: Don't assume getauxval is Linux-only

SerenityOS doesn't expose Linux-compatible hwcaps (like HWCAP_CRC32)
in the auxiliary vector.
---
 src/liblzma/check/crc32_arm64.h | 2 +-
 src/liblzma/check/crc_common.h  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/liblzma/check/crc32_arm64.h b/src/liblzma/check/crc32_arm64.h
index 4a0818f173004fa85dcf489135f8eec6c0e30f02..54f2bd307bff4eb45893f41fb9923867a7a61e76 100644
--- a/src/liblzma/check/crc32_arm64.h
+++ b/src/liblzma/check/crc32_arm64.h
@@ -108,7 +108,7 @@ crc32_arch_optimized(const uint8_t *buf, size_t size, uint32_t crc)
 static inline bool
 is_arch_extension_supported(void)
 {
-#if defined(HAVE_GETAUXVAL) && defined(HAVE_HWCAP_CRC32)
+#if defined(__linux__) && defined(HAVE_GETAUXVAL) && defined(HAVE_HWCAP_CRC32)
 	return (getauxval(AT_HWCAP) & HWCAP_CRC32) != 0;
 
 #elif defined(HAVE_ELF_AUX_INFO)
diff --git a/src/liblzma/check/crc_common.h b/src/liblzma/check/crc_common.h
index 4897dfee49ae02e509eaac7d2dbb60192c53b114..0770e9386d4cd5d860cb596dbd37e86f81804c6b 100644
--- a/src/liblzma/check/crc_common.h
+++ b/src/liblzma/check/crc_common.h
@@ -90,7 +90,7 @@ extern const uint64_t lzma_crc64_table[4][256];
 //
 // Keep this in sync with changes to crc32_arm64.h
 #if defined(_WIN32) \
-		|| (defined(HAVE_GETAUXVAL) && defined(HAVE_HWCAP_CRC32)) \
+		|| (defined(__linux__) && defined(HAVE_GETAUXVAL) && defined(HAVE_HWCAP_CRC32)) \
 		|| defined(HAVE_ELF_AUX_INFO) \
 		|| (defined(__APPLE__) && defined(HAVE_SYSCTLBYNAME))
 #	define CRC_ARM64_RUNTIME_DETECTION 1
